File: CCC-2019/S1.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> grid = {1,2,3,4};

void vertical()
{
    int a = grid[0],b = grid[2];
    grid[0] = grid[1]; grid[2] = grid[3];
    grid[1] = a; grid[3] = b;
};

void horizontal()
{
    int a = grid[2],b = grid[3];
    grid[2] = grid[0]; grid[3] = grid[1];
    grid[0] = a; grid[1] = b;
};

int main()
{
    string seq; cin >> seq;
    for (int i=0;i<seq.length();i++)
    {
        if (seq[i] == 'H') {horizontal();};
        if (seq[i] == 'V') {vertical();};
    };
    cout << grid[0] << ' ' << grid[1] << endl;
    cout << grid[2] << ' ' << grid[3] << endl;
};

File: CCC-2019/Python/S1.py
grid = [1,2,3,4]

def vertical():
    numHolderOne = grid[0]
    numHolderTwo = grid[2]
    grid[0] = grid[1]
    grid[2] = grid[3]
    grid[1] = numHolderOne
    grid[3] = numHolderTwo

def horizontal():
    numHolderOne = grid[2]
    numHolderTwo = grid[3]
    grid[2] = grid[0]
    grid[3] = grid[1]
    grid[0] = numHolderOne
    grid[1] = numHolderTwo

sequence = input()
for i in range(0, len(sequence)):
    if sequence[i] == 'H':
        horizontal()
    else:
        vertical()

print(f'{grid[0]} {grid[1]}')
print(f'{grid[2]} {grid[3]}')
